bluesky/run_engine.py

```diff
@@ -341,6 +342,8 @@ def __init__(self, md=None, *, loop=None, preprocessors=None,
         self.record_interruptions = False
         self.pause_msg = PAUSE_MSG
 
```

Comment:
I hope we can find a way to solve this without introducing this coupling to matplotlib and a `sys.modules` check deeply into RunEngine. Would it work to do it in the __init__ of LivePlot (and similar)?

Reply:
I would feel better if we moved this to the code that sets up the background thread ?  It would still be run on init, but it would at least put all of the funny stuff together?

Reply:
By the code that sets up the background thread you mean `_ensure_event_loop_running` function?

Actually, if all Matplotlib/Qt related code should be moved to `default_during_task`, the function could be changed to a callable class which will have the methods `initialize` (to initializes the teleporter if the backend is Qt) and `__call__`  (the same code as current implementation of `default_during_task`). The method 'initalize' can be called from `RE.__init__()`.

Reply:
I had that same thought, in fact. If this has to be a two-step process for Qt, it might be a two-step process for other implementations as well. Establishing a two-part interface for it seems right.

Reply:
To get into names/details, maybe expect an object with methods `initialize` and `block`. We use `__call__` a lot of places, but we do that partly for brevity in user code and partly so that plain functions can be used in some cases (e.g. for callbacks). Neither of those reasons apply here, so an explicit method like `block` might be more clear.